src/gateway.rs

```diff
@@ -64,29 +65,36 @@ impl MessageSender {
 
 pub struct Gateway {
     uplinks: dispatcher::MessageSender,
+    self_sender: MessageSender,
     messages: MessageReceiver,
     beacon_handler: beaconer::MessageSender,
     downlink_mac: MacAddress,
     udp_runtime: UdpRuntime,
     listen_address: String,
     region_params: Option<RegionParams>,
+    /// If packet forwarder returns InvalidTxPower, it provides the max allowable tx value
+    /// which we can save and apply it as a maximum to outgoing packets
+    max_power: Option<u32>,
 }
 
 impl Gateway {
     pub async fn new(
         uplinks: dispatcher::MessageSender,
+        self_sender: MessageSender,
```

Comment:
this seems like a hack? you're passing in a MessageSender that assumes it's a sender for the given messages receiver

Reply:
Fundamentally, there's two ways of solving the original problem, ie: "How to send InvalidTransmitPower back after being spawned off"

Solution 1: Mutex
Solution 2: Channel

I think interacting directly with mutexes is annoying, so I'd rather extend the use of an existing channel. And I don't think sending messages to my own queue is a hack.

Reply:
_OR_ use another message channel instead of assuming that the self_sender will send to messages